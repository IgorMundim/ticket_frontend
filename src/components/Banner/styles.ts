import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
${({ theme }) => css`
  margin-top:${theme.spacings.xxsmall};
  display: flex;
  align-items: center;
  height: 100%;
  overflow: hidden;
  position: relative;
`}
`;

export const List = styled.div`
  display: flex;

`;

export const Slide = styled.div`
${({ theme }) => css` 
  cursor: grab;
  position: relative;
  width: 90vw;
  flex-shrink: 0;
  flex-direction: column;
  padding: 0 ${theme.spacings.xxsmall};
  border-radius: ${theme.spacings.tiny};

  `}
`;

export const Image = styled.img`
${({ theme }) => css`
  width: 100%;
  height: 30rem;
  border: ${theme.spacings.xxtiny} solid ${theme.colors.white};
  border-radius: ${theme.spacings.tiny};
  box-shadow: #070707 0px 12px 12px;

  @media ${theme.media.lteMedium}{
    height: 15rem;
  }
`}
`;

export const Description = styled.div`
  ${({ theme }) => css`
   /* background: linear-gradient(to top, ${theme.colors.primary}, transparent); */
   position: absolute;
   bottom: 0; 
   padding: ${theme.spacings.mediumSmall};
   
   > h3 {
    position: relative;
    top: ${theme.spacings.tiny};
   }
  `}
`;

export const GoLeft = styled.div`
  ${({ theme }) => css`
    left: ${theme.spacings.xsmall};
    position: absolute;
    z-index: 3;
  `}
`;

export const GoRight = styled.div`
  ${({ theme }) => css`
    right: ${theme.spacings.xsmall};
    position: absolute;
    z-index: 3;
  `}
`;
export const Button = styled.button`
  ${({ theme }) => css`    
    background: transparent;
    color: ${theme.colors.white};
    cursor: pointer;
    width: 2rem;
    height: 2rem;
    border: none;
    z-index: 3;
    
  `}
`;

export const Control = styled.div`    
    z-index: 3;
    padding-top: 0.3rem;
    display: flex;
    justify-content: center;
    align-items: center;

`;
