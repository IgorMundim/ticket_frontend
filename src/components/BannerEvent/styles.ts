import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
      position: relative;
      width: 100%; 
      height: 40vh;
      @media ${theme.media.lteMedium}{
        height: 25vh;
      }
  `} 
`;

export const Image = styled.img`
  height: 100%;
  width: 100%;
  filter: brightness(90%);
`;
