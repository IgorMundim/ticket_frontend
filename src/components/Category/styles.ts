import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  
`;
export const Category = styled.div`
  width: 22rem;
  padding: 0 5rem;
  flex-shrink: 0;
  cursor: pointer;
  
`;
export const Image = styled.img`  
  border-radius: 50%; 
  width: 100%;
  height: 12rem;
`;
export const Description = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const Link = styled.a`
  text-decoration: none;
  color: inherit;
`;
