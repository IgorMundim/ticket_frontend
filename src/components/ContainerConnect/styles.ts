import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    max-width: 50rem;
    margin: 5rem auto;
    backgorund: ${theme.colors.white};
    padding: ${theme.spacings.xlarge};
    > button {
    width:100%;
    margin-bottom: 3rem;

  }
  `}
`;
