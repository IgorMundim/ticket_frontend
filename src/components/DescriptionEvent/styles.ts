import styled from 'styled-components';

export const Container = styled.div`
  > h2 {
    margin-bottom: 3rem;
  }
  > div {
    font-size: 2rem;
    > h1, h2, h3, h4, h5, h6 {
      font-size: 2rem;
      margin: 1.5rem 0;
    }
  }

`;
