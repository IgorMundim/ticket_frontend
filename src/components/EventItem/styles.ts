import styled, { css } from 'styled-components';

export const Container = styled.div`
   
   width: 34rem;
   margin: 0 auto;
   display: grid;
   grid-template-rows: 19rem 1fr ;
`;
export const Detail = styled.div`
   ${({ theme }) => css`
   border-bottom: .3rem solid ${theme.colors.gray2};
   border-left: .3rem solid ${theme.colors.gray2};
   border-right: .3rem solid ${theme.colors.gray2};
   border-radius: 0 0 0.7rem 0.7rem;
   display: grid;
   grid-template-columns: 1fr 4fr;
   `}
`;

export const Description = styled.div`
   padding: 1rem;
   > h3 {   
     word-wrap: break-word;
     cursor: pointer;
  } 
`;

export const Date = styled.div`
   word-wrap: break-word;
   padding: 1rem;
   text-align: center;
`;
export const Image = styled.img`
   border-radius: 0.7rem 0.7rem 0 0;
   width: 100%;
   height: 100%; 
   cursor: pointer;
`;
