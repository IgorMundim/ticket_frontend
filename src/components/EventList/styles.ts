import styled, { css } from 'styled-components';

export const Container = styled.div`
${({ theme }) => css`
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  gap: ${theme.spacings.small};
  
  width:100%;
  @media ${theme.media.lteMedium}{
    grid-template-columns: 1fr;
  }
  `}
`;
