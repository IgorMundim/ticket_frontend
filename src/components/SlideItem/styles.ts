import styled, { css } from 'styled-components';

export const Container = styled.div``;

export const Description = styled.div`
  ${({ theme }) => css`
   /* background: linear-gradient(to top, ${theme.colors.primary}, transparent); */
   position: absolute;
   bottom: 0; 
   padding: ${theme.spacings.mediumSmall};
   
   > h3 {
    position: relative;
    top: ${theme.spacings.tiny};
   }
  `}
`;

export const Slide = styled.div`
${({ theme }) => css` 
  cursor: grab;
  position: relative;
  width: 90vw;
  flex-shrink: 0;
  flex-direction: column;
  padding: 0 ${theme.spacings.xxsmall};
  border-radius: ${theme.spacings.tiny};

  `}
`;
