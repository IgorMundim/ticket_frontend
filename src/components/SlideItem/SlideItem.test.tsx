import { screen } from '@testing-library/react';
import { SlideItem } from '.';
import { renderTheme } from '../../styles/render-theme';
describe('<SlideItem />', () => {
  it('should render slide item', () => {
    const { container } = renderTheme(
      <SlideItem
        src="img.jpg"
        title="What is Lorem Ipsum"
        description="There is no one who loves pain itself"
        alt="text"
        link="https://www.google.com/"
      />,
    );
    expect(
      screen.getAllByRole('heading', {
        name: 'What is Lorem Ipsum',
      }),
    );

    expect(container).toMatchSnapshot();
  });
});
