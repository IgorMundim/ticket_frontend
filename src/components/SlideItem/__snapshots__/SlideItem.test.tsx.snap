// Vitest Snapshot v1

exports[`<SlideItem /> > should render slide item 1`] = `
.c4 {
  color: #FFFFFF;
  font-size: 1.2rem;
  text-transform: none;
}

.c1 {
  width: 100%;
  height: 30rem;
  border: 0.1rem solid #FFFFFF;
  border-radius: 0.4rem;
  box-shadow: #070707 0px 12px 12px;
}

.c3 {
  color: #FFFFFF;
  font-size: 1.6rem;
  text-transform: uppercase;
}

.c2 {
  position: absolute;
  bottom: 0;
  padding: 2.0rem;
}

.c2 > h3 {
  position: relative;
  top: 0.4rem;
}

.c0 {
  cursor: -webkit-grab;
  cursor: -moz-grab;
  cursor: grab;
  position: relative;
  width: 90vw;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0 0.8rem;
  border-radius: 0.4rem;
}

@media (max-width:768px) {
  .c1 {
    height: 15rem;
  }
}

<div>
  <div
    class="c0"
  >
    <img
      alt="text"
      class="c1"
      src="img.jpg"
    />
    <div
      class="c2"
    >
      <h3
        class="c3"
      >
        What is Lorem Ipsum
      </h3>
      <p
        class="c4"
      >
        There is no one who loves pain itself
      </p>
    </div>
  </div>
</div>
`;
